Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -15,23 +15,42 @@
 import AsyncHTTPClient
 import AWSSignerV4
 import AWSXML
+import Baggage
 import Dispatch
 import struct Foundation.Data
 import class Foundation.JSONDecoder
 import class Foundation.JSONSerialization
 import struct Foundation.URL
 import struct Foundation.URLQueryItem
+import Instrumentation
 import Logging
 import Metrics
 import NIO
 import NIOConcurrencyHelpers
 import NIOHTTP1
 import NIOTransportServices
+import TracingInstrumentation
+
+// TODO: instrument initialization
+// TODO: create internal spans for signing, parsing?
+
+// TODO: remove and use optional parameter or dont provide default value
+public struct AWSClientContext: AWSClient.Context {
+    public static var empty: AWSClientContext = .init(baggage: .init())
+
+    public var baggage: BaggageContext
+}
 
 /// This is the workhorse of aws-sdk-swift-core. You provide it with a `AWSShape` Input object, it converts it to `AWSRequest` which is then converted
 /// to a raw `HTTPClient` Request. This is then sent to AWS. When the response from AWS is received if it is successful it is converted to a `AWSResponse`
 /// which is then decoded to generate a `AWSShape` Output object. If it is not successful then `AWSClient` will throw an `AWSErrorType`.
 public final class AWSClient {
+    /// AWS operation context including trace context
+    public typealias Context = Baggage.BaggageContextCarrier
+    // TODO: import BaggageLogging and add confirmance to LoggingBaggageContextCarrier
+    // when naming and conventions are confirmed https://github.com/slashmo/gsoc-swift-baggage-context/issues/23
+    // remove logger parameter (it will be provided in the context)
```

Comment:
the context may also be logger carrier, discussion here https://github.com/slashmo/gsoc-swift-baggage-context/issues/23